src/platform/posix/include/platform/platform.h

```diff
@@ -24,6 +24,8 @@
 
 #define PLATFORM_DEFAULT_DELAY 12
 
+struct sof;
```

Comment:
`extern`?

Reply:
I think this is ok as a forward declaration.

Reply:
IIRC there is a weird (IMHO) C feature, that yes, you can use this in a header and the compiler will "guess what you actually mean" and only create one instance of the object, but I'd rather use an explicit `extern` the same way as we do everywhere, particularly since at least me personally I'm not sure which C standard we settled with and from which version that feature was introduced

Reply:
There is a very good description of that portability issue at the end of the chapter "Declarations" in the Harbison & Steele (which every C developer should have IMHO). For maximum portability across various toolchains (which I think matters to SOF), they definitely agree with @lyakh .

Follow-up PR? Could also fix some other checkpatch warnings inhttps://github.com/thesofproject/sof/actions/runs/4118174248/jobs/7110383634

Reply:
humm, the use of 'struct foo' is very very common in Linux headers. I'd be surprised if you saw any 'extern' in any of the ALSA/ASoC headers...

Reply:
FWIW, this is a type declaration, not an object definition.  All it means is that "struct sof exists as a type and you can legally take a pointer to it".  You see this construct in C++ a lot, because it avoids the need to pull in the very expensive headers that define types.   The reason it was introduced here is that I had a spot further down where I declare a function that takes a "struct sof *" as an argument, and was too lazy to figure out where that was declared.

Reply:
> FWIW, this is a type declaration, not an object definition.

Thanks @andyross , I read too fast and stand corrected (and I think you meant: "not an _object_... declaration" :-)

Reply:
> FWIW, this is a type declaration, not an object definition.

ouch... /me looking at those "computer glasses" and contemplates if he should wear them more frequently... Then hides away

Reply:
no, of course it's a perfectly fine forward type declaration. And I myself frequently suggest it when just a pointer to that type is needed - exactly as @andyross explained, and I think that's usually preferred over including a header just for that. And that's also why it's used in Linux, in SOF, etc. I just somehow misread it as `struct sof sof;`...